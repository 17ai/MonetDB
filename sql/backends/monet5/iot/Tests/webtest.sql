set schema iot;
set optimizer='iot_pipe';
create stream table ntemps( iotclk timestamp, room string , temperature real);
create table atemps( iotclk timestamp, cnt int , temperature real);
-- remainder depends on location of the baskets root

declare baskets string;
set baskets= '/ufs/mk/baskets/measures/temperatures/';

call iot.import('iot','ntemps', concat(baskets,'1'));
select * from ntemps;

create procedure web00()
begin
    insert into atemps select min(iotclk), count(*), avg(temperature) from ntemps;
end;

call iot.query('iot','web00');
call iot.show('iot','web00');

call iot.stop();
--select * from  iot.baskets();
--select * from  iot.queries();

