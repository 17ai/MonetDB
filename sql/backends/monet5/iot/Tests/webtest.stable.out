stdout of test 'webtest` in directory 'sql/backends/monet5/iot` itself:


# 22:26:06 >  
# 22:26:06 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=31285" "--set" "mapi_usock=/var/tmp/mtest-10345/.s.monetdb.31285" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 22:26:06 >  

# MonetDB 5 server v11.24.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5_iot', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2016 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:31285/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-10345/.s.monetdb.31285
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/iot loaded

Ready.

# 22:26:06 >  
# 22:26:06 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-10345" "--port=31285"
# 22:26:06 >  

#set schema iot;
#create stream table temps( iotclk timestamp, room string , temperature real);
#create table atemps( iotclk timestamp, cnt int , temperature real);
#declare baskets string;
#set baskets= '/ufs/mk/baskets/measures/temperatures/';
#create procedure web00()
#begin
#    insert into atemps select min(iotclk), count(*), avg(temperature) from temps;
#end;
#select * from  iot.baskets();
% .,	.,	.,	.,	.,	.,	.,	.,	.,	.,	. # table_name
% schema,	table,	status,	threshold,	winsize,	winstride,	timeslice,	timestride,	beat,	seen,	events # name
% clob,	clob,	clob,	int,	int,	int,	int,	int,	int,	timestamp,	int # type
% 3,	5,	6,	1,	1,	1,	1,	1,	1,	26,	1 # length
[ "iot",	"temps",	"active",	0,	0,	0,	0,	0,	0,	NULL,	5	]
#select * from  iot.queries();
% .,	.,	.,	.,	.,	.,	.,	. # table_name
% schema,	function,	status,	lastrun,	cycles,	events,	time,	error # name
% clob,	clob,	clob,	timestamp,	int,	int,	bigint,	clob # type
% 3,	5,	6,	26,	1,	1,	1,	0 # length
[ "iot",	"web00",	"paused",	NULL,	0,	0,	0,	""	]

# 22:26:06 >  
# 22:26:06 >  "Done."
# 22:26:06 >  

